service/src/main/java/bio/terra/workspace/service/policy/TpsApiDispatch.java

```diff
@@ -0,0 +1,208 @@
+import bio.terra.policy.model.TpsPaoUpdateResult;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.policy.model.TpsUpdateMode;
+import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
+import bio.terra.workspace.app.configuration.external.PolicyServiceConfiguration;
+import bio.terra.workspace.service.policy.exception.PolicyServiceAPIException;
+import bio.terra.workspace.service.policy.exception.PolicyServiceAuthorizationException;
+import bio.terra.workspace.service.policy.exception.PolicyServiceDuplicateException;
+import bio.terra.workspace.service.policy.exception.PolicyServiceNotFoundException;
+import io.opencensus.contrib.http.jaxrs.JaxrsClientExtractor;
+import io.opencensus.contrib.http.jaxrs.JaxrsClientFilter;
+import io.opencensus.contrib.spring.aop.Traced;
+import io.opencensus.trace.Tracing;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.ws.rs.client.Client;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TpsApiDispatch {
+  private final FeatureConfiguration features;
+  private final PolicyServiceConfiguration policyServiceConfiguration;
+  private final Client commonHttpClient;
+
+  @Autowired
+  public TpsApiDispatch(
+      FeatureConfiguration features, PolicyServiceConfiguration policyServiceConfiguration) {
+    this.features = features;
+    this.policyServiceConfiguration = policyServiceConfiguration;
+    this.commonHttpClient =
+        new ApiClient()
+            .getHttpClient()
+            .register(
+                new JaxrsClientFilter(
+                    new JaxrsClientExtractor(), Tracing.getPropagationComponent().getB3Format()));
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createPao(UUID workspaceUuid, @Nullable TpsPolicyInputs policyInputs) {
+    features.tpsEnabledCheck();
+    TpsPolicyInputs inputs = (policyInputs == null) ? new TpsPolicyInputs() : policyInputs;
+
+    TpsApi tpsApi = policyApi();
+    try {
+      tpsApi.createPao(
+          new TpsPaoCreateRequest()
+              .objectId(workspaceUuid)
+              .component(TpsComponent.WSM)
+              .objectType(TpsObjectType.WORKSPACE)
+              .attributes(inputs));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  public void deletePao(UUID workspaceUuid) {
+    features.tpsEnabledCheck();
+    TpsApi tpsApi = policyApi();
+    try {
+      try {
+        tpsApi.deletePao(workspaceUuid);
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found is not an error as far as WSM is concerned.
+    }
+  }
+
+  @Traced
```

Comment:
out of curiosity, why is only this endpoint traced?

Reply:
Good point. All of the external calls should be traced.